app/configurator/components/filters.tsx

```diff
@@ -533,31 +533,32 @@ const useEnsureUpToDateColorMapping = ({
 
   useEffect(() => {
     if (
-      activeField &&
       hasOutdatedMapping &&
       colorMapping &&
-      colorComponentValues
+      colorComponentValues &&
+      activeField
     ) {
       dispatch({
         type: "CHART_CONFIG_UPDATE_COLOR_MAPPING",
         value: {
           dimensionId,
           colorConfigPath,
           colorMapping,
-          field: activeField,
+          field: isColorInConfig(chartConfig) ? "color" : activeField,
```

Comment:
@noahonyejese I'd extract this outside of the hook, see that now we have `chartConfig` in dependencies, and it will re-run every time we change anything in `chartConfig`, while we should only re-run if `field` changes. Does it make sense? :)